src/main/java/com/fasterxml/jackson/core/ErrorReportConfiguration.java

```diff
@@ -0,0 +1,191 @@
+package com.fasterxml.jackson.core;
+
+import java.io.Serializable;
+
+/**
+ * Container for configuration values used when handling errorneous token inputs. 
+ * For example, unquoted text segments.
+ * <p>
+ * Currently default settings are
+ * <ul>
+ *     <li>Maximum length of token to include in error messages (see {@link #_maxErrorTokenLength})
+ *     <li>Maximum length of raw content to include in error messages (see {@link #_maxRawContentLength})
+ * </ul>
+ *
+ * @since 2.16
+ */
+public class ErrorReportConfiguration
+        implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Default value for {@link #_maxErrorTokenLength}.
+     */
+    public static final int DEFAULT_MAX_ERROR_TOKEN_LENGTH = 256;
+
+    /**
+     * Previous was {@link com.fasterxml.jackson.core.io.ContentReference#DEFAULT_MAX_CONTENT_SNIPPET}.
+     * Default value for {@link #_maxRawContentLength}.
+     */
+    public static final int DEFAULT_MAX_RAW_CONTENT_LENGTH = 500;
+    
+    /**
+     * Maximum length of token to include in error messages
+     *
+     * @see Builder#maxErrorTokenLength(int)
+     */
+    protected final int _maxErrorTokenLength;
+
+    /**
+     * Maximum length of raw content to include in error messages
+     * 
+     * @see Builder#maxRawContentLength(int) 
+     */
+    protected final int _maxRawContentLength;
+
+    private static ErrorReportConfiguration DEFAULT =
+            new ErrorReportConfiguration(DEFAULT_MAX_ERROR_TOKEN_LENGTH, DEFAULT_MAX_RAW_CONTENT_LENGTH);
+
+    public static void overrideDefaultErrorReportConfiguration(final ErrorReportConfiguration errorReportConfiguration) {
```

Comment:
Could you copy Javadoc from matching method in `StreamWriteConstaints`, warning against use of this method?
In general it should not be used and specifically libraries should not change global defaults, ever.

Reply:
Or actually I can do this.